server/routerlicious/packages/services-shared/src/startupChecker.ts

```diff
@@ -0,0 +1,26 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+export class StartupChecker {
```

Comment:
It feels like this should be an implementation of `IReadinessCheck` with `isReady()` rather than `isStartupComplete()`. That will more easily allow alternative implementations to be passed in.

Reply:
I think we kept the `StartupChecker` implementation independent of the `ReadinessCheck`. We wanted this as the startup check is pretty straight forward that only works to ensure that the resources and runner are setup. This implementation imo does not need much customization that readiness would require. Hence, I did not implement this class to be customizable.